[mlir][mlir-vulkan-runner] Move part of device pass pipeline to mlir-opt

andfau-amd opened the pull request:
Adds a new mlir-opt test-only pass, -test-vulkan-runner-pipeline, which runs a set of passes needed for mlir-vulkan-runner, and removes them from the runner. The tests are changed to invoke mlir-opt with this flag before invoking the runner. The passes moved are ones concerned with lowering of the device code prior to serialization to SPIR-V. This is an incremental step towards moving the entire pipeline to mlir-opt, to align with other runners (see #73457).

joker-eph commented:
Thanks, that LGTM, but let's wait for @kuhar to review.

kuhar commented on mlir/test/lib/Pass/TestVulkanRunnerPipeline.cpp:
```diff
@@ -0,0 +1,49 @@
+
+} // namespace
+
+namespace mlir {
+namespace test {
```
```suggestion
namespace mlir::test {
```

kuhar commented on mlir/test/mlir-vulkan-runner/addf.mlir:
```diff
@@ -1,4 +1,5 @@
-// RUN: mlir-vulkan-runner %s --shared-libs=%vulkan-runtime-wrappers,%mlir_runner_utils --entry-point-result=void | FileCheck %s
+// RUN: mlir-opt %s -test-vulkan-runner-pipeline \
+// RUN: | mlir-vulkan-runner - --shared-libs=%vulkan-runtime-wrappers,%mlir_runner_utils --entry-point-result=void | FileCheck %s
```
```suggestion
// RUN:   | mlir-vulkan-runner - --shared-libs=%vulkan-runtime-wrappers,%mlir_runner_utils --entry-point-result=void | FileCheck %s
```

kuhar replied:
Also in the other files

kuhar reviewed (approved):
LGTM % minor formatting issues

andfau-amd committed:
[mlir][mlir-vulkan-runner] Move part of device pass pipeline to mlir-opt

Adds a new mlir-opt test-only pass, -test-vulkan-runner-pipeline, which
runs a set of passes needed for mlir-vulkan-runner, and removes them
from the runner. The tests are changed to invoke mlir-opt with this flag
before invoking the runner. The passes moved are ones concerned with
lowering of the device code prior to serialization to SPIR-V. This is an
incremental step towards moving the entire pipeline to mlir-opt, to
align with other runners (see #73457).
